docs/src/pages/demos/autocomplete/IntegrationReactSelect.tsx

```diff
@@ -0,0 +1,290 @@
+    fontSize: 16,
+  },
+  placeholder: {
+    position: 'absolute',
+    left: 2,
+    fontSize: 16,
+  },
+  paper: {
+    position: 'absolute',
+    zIndex: 1,
+    marginTop: theme.spacing(1),
+    left: 0,
+    right: 0,
+  },
+  divider: {
+    height: theme.spacing(2),
+  },
+});
+
+function NoOptionsMessage(props: NoticeProps<OptionType>) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.noOptionsMessage}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  );
+}
+
+type InputComponentProps = Pick<BaseTextFieldProps, 'inputRef'> & HTMLAttributes<HTMLDivElement>;
+
+function inputComponent({ inputRef, ...props }: InputComponentProps) {
+  return <div ref={inputRef} {...props} />;
+}
+
+function Control(props: ControlProps<OptionType>) {
+  return (
+    <TextField
+      fullWidth
+      InputProps={{
+        inputComponent,
+        inputProps: {
+          className: props.selectProps.classes.input,
+          inputRef: props.innerRef,
+          children: props.children,
+          ...props.innerProps,
+        },
+      }}
+      {...props.selectProps.textFieldProps}
+    />
+  );
+}
+
+function Option(props: OptionProps<OptionType>) {
+  return (
+    <MenuItem
+      buttonRef={props.innerRef}
+      selected={props.isFocused}
+      component="div"
+      style={{
+        fontWeight: props.isSelected ? 500 : 400,
+      }}
+      {...props.innerProps}
+    >
+      {props.children}
+    </MenuItem>
+  );
+}
+
+function Placeholder(props: PlaceholderProps<OptionType>) {
+  return (
+    <Typography
+      color="textSecondary"
+      className={props.selectProps.classes.placeholder}
+      {...props.innerProps}
+    >
+      {props.children}
+    </Typography>
+  );
+}
+
+function SingleValue(props: SingleValueProps<OptionType>) {
+  return (
+    <Typography className={props.selectProps.classes.singleValue} {...props.innerProps}>
+      {props.children}
+    </Typography>
+  );
+}
+
+function ValueContainer(props: ValueContainerProps<OptionType>) {
+  return <div className={props.selectProps.classes.valueContainer}>{props.children}</div>;
+}
+
+function MultiValue(props: MultiValueProps<OptionType>) {
+  return (
+    <Chip
+      tabIndex={-1}
+      label={props.children}
+      className={clsx(props.selectProps.classes.chip, {
+        [props.selectProps.classes.chipFocused]: props.isFocused,
+      })}
+      onDelete={props.removeProps.onClick}
+      deleteIcon={<CancelIcon {...props.removeProps} />}
+    />
+  );
+}
+
+function Menu(props: MenuProps<OptionType>) {
+  return (
+    <Paper square className={props.selectProps.classes.paper} {...props.innerProps}>
+      {props.children}
+    </Paper>
+  );
+}
+
+const components = {
+  Control,
+  Menu,
+  MultiValue,
+  NoOptionsMessage,
+  Option,
+  Placeholder,
+  SingleValue,
+  ValueContainer,
+};
+
+class IntegrationReactSelect extends React.Component<WithStyles<string, true>> {
```

Comment:
Please follow our typescript guide concerning `WithStyles` usage.

Reply:
You mean `WithStyles<typeof styles, true>` ?

The guide talk about
```
interface Props extends WithStyles<typeof styles> {
  foo: number;
  bar: boolean;
}
```
But since there is no props, it's useless to create
```
interface Props extends WithStyles<typeof styles> {}
```

Reply:
> You mean WithStyles<typeof styles, true> ?

Yes. Although I personally would encourage to type out the interface. Any patch that adds props will reduce the amount of diff by only including the prop line and not the refactor from inline props to interface props.

Reply:
I did the change.

Actually the tests failed cause (of course) I have not the same code for `.js` and `.tsx` since I disable more eslint rules and use `ts-ignore` ...

Since the demo looks fine to you ; what about the more important part: the typing bug fix ?